### PR TITLE
[innogysmarthome] Reconnect fixes (#8182)

### PR DESCRIPTION
**Pull request by Novanic**
This pull-request fixes a few reconnect problems regarding the WebSocket, see issue #8182 for more information
- When an error occurred while starting the websocket, no reconnect attempt was started which caused that the binding loses the connection (till the next restart of the server or binding), see issue #8182 for more information
- When an error occurs, it is now waited at least for 30 seconds for the next reconnect attempt to avoid too many requests (on re-occurring errors).
- Fix for a side-effect of "Fix possible resource leak #8080" (the websocket were closed unnormally by default which could trigger false and about one reconnect attempt per second).
- NullPointerException fixed

### CODE REVIEW

File: bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/InnogyWebSocket.java [hunk 1]
```diff
@@ -128,7 +128,8 @@ public void onConnect(Session session) {
     public void onClose(int statusCode, String reason) {
         if (statusCode == StatusCode.NORMAL) {
             logger.info("Connection to innogy Webservice was closed normally.");
-        } else {
+        } else if(!closing) {
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/InnogyWebSocket.java:131]:
```suggestion
        } else if (!closing) {
```

**Review by Hilbrand — APPROVED** [review]
LGTM. only a small style comment, which I'll fix with GitHub commit suggestion.

**Comment by Novanic** [comment]
@Hilbrand Please revert the change again... :-( I don't understand what's going on with this binding... I tested it for weeks and now since it is merged I get rapid (re-)connect attempts again...

Please revert it shortly before the release of 2.5.8 (which may be built tomorrow?!).

**Comment by Novanic** [comment]
Revert pull-request: https://github.com/openhab/openhab-addons/pull/8324

**Pull request merged**